app/jobs/cancel_reservations_job.rb

```diff
@@ -0,0 +1,23 @@
+class CancelReservationsJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    reservations.each do |reservation|
+      cancel_reservation(reservation) if not_confirmed_on_time?(reservation)
+    end
+  end
+
+  private
+
+  def not_confirmed_on_time?(reservation)
+    reservation.seance.start_time <= 30.minutes.from_now
```

Comment:
Are you sure that this condition is valid? :wink: 

It looks like the most crucial thing in this PR, why don't you want to cover it by unit test?

Reply:
Yup, it's valid. I wrote 3 tests for start_time being less than, equal, and more than 30 second :D.